unit_tests/src/test_connection_pool.cpp

```diff
@@ -379,6 +379,108 @@ TEST_CASE("connection pool")
             REQUIRE_NOTHROW(adm::client::modify_resource(conn, "demoResc", adm::resource_comments_property{""}));
         }
 
+        SECTION("refresh connection after adding/removing resource")
+        {
+            cp_opts.refresh_connections_when_resource_changes_detected = true;
+
+            irods::connection_pool conn_pool{1, env.rodsHost, env.rodsPort, user, cp_opts};
+
+            namespace adm = irods::experimental::administration;
+
+            RcComm* conn_ptr{};
+
+            const char* resc_name = "unit_test_repl_tracked";
+
+            {
+                auto conn = conn_pool.get_connection();
+                REQUIRE(conn);
+
+                conn_ptr = static_cast<RcComm*>(conn);
+                REQUIRE(conn_ptr);
+
+                adm::resource_registration_info repl_info;
+                repl_info.resource_name = resc_name;
+                repl_info.resource_type = adm::resource_type::replication;
+
+                // This will cause all connections in the pool to be replaced on their next usage.
+                REQUIRE_NOTHROW(adm::client::add_resource(conn, repl_info));
+            }
+
+            {
+                auto conn = conn_pool.get_connection();
+                REQUIRE(conn);
+                REQUIRE(static_cast<RcComm*>(conn) != conn_ptr);
+
+                // Capture the memory address again. We'll need this later.
+                conn_ptr = static_cast<RcComm*>(conn);
+
+                // This will cause all connections in the pool to be replaced on their next usage.
+                REQUIRE_NOTHROW(adm::client::remove_resource(conn, resc_name));
+            }
+
+            // Show the connection in the pool has been updated again.
+            auto conn = conn_pool.get_connection();
+            REQUIRE(conn);
+            REQUIRE(static_cast<RcComm*>(conn) != conn_ptr);
+        }
+
+        SECTION("refresh connection after resource is modified by external source")
```

Comment:
I guess it's a form of modification, but should this read like this?

```suggestion
        SECTION("refresh connection after resource is removed by external source")
```

Reply:
Will tweak to be more specific about what is happening.